packages/idempotency/src/IdempotencyHandler.ts

```diff
@@ -275,8 +276,9 @@ export class IdempotencyHandler<Func extends AnyFunction> {
       !this.#idempotencyConfig.throwOnNoIdempotencyKey
     ) {
       const selection = search(
+        this.#idempotencyConfig.eventKeyJmesPath,
         this.#functionPayloadToBeHashed,
-        this.#idempotencyConfig.eventKeyJmesPath
+        { customFunctions: new PowertoolsFunctions() }
```

Comment:
Is there any reason to create new object every time? This pattern goes doen to `IdempotencyHandler`. I am wondering if we can use this as a private field, this object is the same everywhere, we can use it once.